### PR TITLE
Use Mopidy's command API for initial auth

### PR DESCRIPTION
**Pull request by jacobobryant**
This will provide a `mopidy gmusic login` command for getting the refresh token.

### CODE REVIEW

**Comment by jacobobryant** [comment]
I didn't update the readme fyi, so that would need to be done at some point (after the next release to pypi?).

**Comment by belak** [comment]
I like this! Thanks for taking the time to submit it. I'm super busy for the next month or so but I'll try and do what I can.

**Comment by jacobobryant** [comment]
Sounds good! Happy to make changes/help in any way I can, just let me know. I see the CI tests failed, though it looks like the tests are just out of date (?).

**Comment by belak** [comment]
There's some flake8 stuff to clean up that seems related to this change, but yes... the tests are broken currently.

https://travis-ci.org/mopidy/mopidy-gmusic/jobs/567119009

File: mopidy_gmusic/__init__.py [hunk 1]
```diff
@@ -21,7 +21,6 @@ def get_default_config(self):
     def get_config_schema(self):
         schema = super(GMusicExtension, self).get_config_schema()
 
-        schema['initial_code'] = config.Secret(optional=True)
```
Comment (jodal) [inline mopidy_gmusic/__init__.py]: This should not be removed, as that will crash any Mopidy installation that has the value set in its configuration file. Instead, change from `config.Secret(...)` to `config.Deprecated()`, and Mopidy will ignore the value when present in the config file.

File: mopidy_gmusic/commands.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+from __future__ import print_function
+from oauth2client.client import OAuth2WebServerFlow
+import gmusicapi
+from mopidy import commands
```
Comment (jodal) [inline mopidy_gmusic/commands.py]: We usually have an empty line in-between each import block. Future imports are one block. stdlib are one block. Each third-party library is one block. The current project is one block. In other words, an empty line in-between each of the four import lines here.

In the future, we'll hopefully have this automated with Black and isort, so it doesn't have be a topic in reviews.

File: mopidy_gmusic/commands.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    def __init__(self):
+        super(GMusicCommand, self).__init__()
+        self.add_child('login', LoginCommand())
+
+
+class LoginCommand(commands.Command):
+    def run(self, args, config):
+        oauth_info = gmusicapi.Mobileclient._session_class.oauth
+        flow = OAuth2WebServerFlow(**oauth_info._asdict())
+        print('')
```
Comment (jodal) [inline mopidy_gmusic/commands.py]:
```suggestion
        print()
```

This provides the exact same behavior.

File: mopidy_gmusic/__init__.py (hunk 1 above)
Comment (belak) [reply mopidy_gmusic/__init__.py]: Is this really an issue if there was never an actual release with that setting?

Comment (jodal) [reply mopidy_gmusic/__init__.py]: No, in that case, I don't think this is an issue at all.

File: README.rst [hunk 4]
```diff
@@ -39,11 +39,11 @@ Install the Mopidy-GMusic extension by running::
 Configuration
 =============
 
-When you first start mopidy, you will see an error along with a URL to visit.
-Once you've logged in to Google through that URL, simply copy the given value
-into the "initial_code" value of the configuration file. On the next run,
-mopidy will get a refresh token for the given code and output it to the
-console. This must then be copied into the config file as the refresh_token.
+Run `mopidy gmusic login` to obtain a refresh token, and then include it in
```
Comment (jodal) [inline README.rst]: Use double backticks to get monospace formatting in reStructuredText.

**Comment by jacobobryant** [comment]
Question about the `deviceid` config key (and the block explaining it in the README). I didn't have to specify a `deviceid` (I've also verified that playback works), and `mopidy-gmusic` didn't output a list of registered devices either (as the README says it would). Is the device id bit just out of date and should be removed?

**Comment by jacobobryant** [comment]
Doh, thought the merge button might show up after hitting "close pull request". Obviously I don't have access to the merge button since it's not my repo ;)

**Comment by mrki0620** [comment]
Hi to all.
I'm trying to perform the initial authentication to my google music account with the mopidy-gmusic extension. Do I understand the steps correctly?:
1) Using mopidy as a service I should first install the mopidy-gmusic extension in context of a "mopidy" user and take the master branch version (where the "mopidy gmusic login" command for getting the refresh token is provided)
sudo -u mopidy pip install https://github.com/mopidy/mopidy-gmusic/archive/master.zip

2) Add configuration to /etc/mopidy/mopidy.conf (what should than be configuration settings added?)
[gmusic]
deviceid = 0123456789abcdef
deviceid = mac
all_access = true
...
3) Run the "mopidy gmusic login" command or (sudo mopidyctl gmusic login)
What are the correct steps to get this working?
Many thanks for your help

**Comment by jacobobryant** [comment]
@mrki0620 looks from the other thread like you got this figured out already--but FYI the `mopidy gmusic login` command won't be on the master branch till this PR is merged. In the mean time, the command could be used by installing the branch in my fork (`pip install https://github.com/jacobobryant/mopidy-gmusic/archive/master.zip`)

**Comment by jodal** [comment]
Sorry we left this hanging! This is a really nice improvement, exactly the kind of thing the Commands API is intended for. Let's get this released quickly :-)

**Comment by jacobobryant** [comment]
Thanks!

**Pull request merged**